examples/asr_sample/utils.js

```diff
@@ -216,6 +226,83 @@ class Utils {
     console.log("Convert output data to ark file.");
   }
 
+  initError(error) {
+    error.numScores = 0,
+    error.numErrors = 0,
+    error.threshold = 0.0001,
+    error.maxError = 0.0,
+    error.rmsError = 0.0,
+    error.sumError = 0.0,
+    error.sumRmsError = 0.0,
+    error.sumSquaredError = 0.0,
+    error.maxRelError = 0.0,
+    error.sumRelError = 0.0,
+    error.sumSquaredRelError = 0.0
+  }
+
+  compareScores(outputTensor, referenceTensor, numRows, numColumns) {
+    let numErrors = 0;
+    this.initError(this.frameError);
+    for (let i = 0; i < numRows; i ++) {
+      for (let j = 0; j < numColumns; j ++) {
+        let score = outputTensor[i*numColumns+j];
+        let refScore = referenceTensor[i * numColumns + j];
+        let error = Math.abs(refScore - score);
+        let rel_error = error / ((Math.abs(refScore)) + 1e-20);
+        let squared_error = error * error;
+        let squared_rel_error = rel_error * rel_error;
+        this.frameError.numScores ++;
+        this.frameError.sumError += error;
+        this.frameError.sumSquaredError += squared_error;
+        if (error > this.frameError.maxError) {
+          this.frameError.maxError = error;
+        }
+        this.frameError.sumRelError += rel_error;
+        this.frameError.sumSquaredRelError += squared_rel_error;
+        if (rel_error > this.frameError.maxRelError) {
+          this.frameError.maxRelError = rel_error;
+        }
+        if (error > this.frameError.threshold) {
```

Comment:
OpenVINO GNA use Int8/Int16 to inference the model, maybe the threshold should be 1 for GNA, the below are result run by native speech sample
```
         max error: 0.705184
         avg error: 0.0448388
         avg rms error: 0.0574098
         stdev error: 0.0371649

```

Reply:
Now just defined the value of threshold according to the [native definition](https://github.com/opencv/dldt/blob/2020/inference-engine/samples/speech_sample/main.cpp#L32). And I checked that when computing errors, native code uses float32 [here](https://github.com/opencv/dldt/blob/2020/inference-engine/samples/speech_sample/main.cpp#L224).

Reply:
> Now just defined the value of threshold according to the native definition. 

The native sample does't use the error num.

> And I checked that when computing errors, native code uses float32 here.

It is output data type not inference precision.

Reply:
Yes, I know that. So should we delete the "error num" in our code or set the threshold to be 1?

Reply:
It's better to use the current value for float32 inference precision, and 1 for int8/int16 precision.

Reply:
@mingmingtasd Do you have any update on the threshold?

Reply:
It's better to update together after refactoring.